Lombiq.HelpfulLibraries.SourceGenerators/Lombiq.HelpfulLibraries.SourceGenerators/ConstantFromJsonGenerator.cs

```diff
@@ -0,0 +1,280 @@
+            var semanticModel = compilation.GetSemanticModel(classDeclarationSyntax.SyntaxTree);
+
+            // Symbols allow us to get the compile-time information.
+            if (semanticModel.GetDeclaredSymbol(classDeclarationSyntax, cancellationToken: context.CancellationToken)
+                is not INamedTypeSymbol classSymbol)
+            {
+                continue;
+            }
+
+            string namespaceName = classSymbol.ContainingNamespace.ToDisplayString();
+
+            // 'Identifier' means the token of the node. Get class name from the syntax node.
+            string className = classDeclarationSyntax.Identifier.Text;
+
+            var partialBody = new StringBuilder();
+
+            // It's possible that a single class is annotated with our marker attribute multiple times
+            foreach (var dictionary in attributeData)
+            {
+                // Get values from dictionary
+                string? constantName = dictionary["constantName"];
+                string? fileName = dictionary["fileName"];
+                string? jsonPath = dictionary["jsonPath"];
+
+                // Try get content of file from dictionary where key ends with filename
+                var fileContent = _fileContents
+                    .FirstOrDefault(k =>
+                        k.Key.EndsWith(fileName.Replace($"\"", string.Empty), StringComparison.Ordinal));
+
+                // If the file content is empty, skip
+                if (string.IsNullOrEmpty(fileContent.Value))
+                {
+                    return;
+                }
+
+                var jsonDocument = JsonDocument.Parse(fileContent.Value);
+
+                // try to find the value in the jsonDocument
+                var jsonValue = FindProperty(jsonDocument.RootElement, jsonPath.Replace("\"", string.Empty));
+
+                if (jsonValue == null)
+                {
+                    return;
+                }
+
+                partialBody.AppendLine($"public const string {constantName.Replace("\"", string.Empty)} = \"{jsonValue.Value}\";");
+            }
+
+            // Create a new partial class with the same name as the original class.
+            // Build up the source code
+            string code = $@"// <auto-generated/>
+
+using System;
+using System.Collections.Generic;
+
+namespace {namespaceName};
+
+partial class {className}
+{{
+    {partialBody}
+}}
+";
+            // Add the source code to the compilation.
+            context.AddSource($"{className}.g.cs", SourceText.From(code, Encoding.UTF8));
+        }
+    }
+
+    /// <summary>
+    ///     Find a property in a JSON document recursively.
+    /// </summary>
+    /// <param name="element">The JSON element to search in.</param>
+    /// <param name="propertyName">The property name to look for.</param>
+    private static JsonElement? FindProperty(JsonElement element, string propertyName)
```

Comment:
This isn't [JSONPath](https://www.ietf.org/archive/id/draft-goessner-dispatch-jsonpath-00.html) compliant, but I see neither is [OrchardCore's STJ implementation](https://github.com/OrchardCMS/OrchardCore/blob/11dc9712403ace4489ea90aa26d7a4c1a78fe0d7/src/OrchardCore/OrchardCore.Abstractions/Json/Nodes/JObject.cs#L91). In this case, please rename the attribute's `jsonPath` to `propertyName` to avoid confusion (and future compatibility issues, in case we ever have to implement full JSONPath support).